ops/pebble.py

```diff
@@ -1308,7 +1308,7 @@ class NoticesSelect(enum.Enum):
     """
 
 
-@dataclasses.dataclass(frozen=True)
+@dataclasses.dataclass
```

Comment:
Yeah, this is good push-back. I changed it because of the implementation of `pebble_notify` / `_notify`, which corresponded directly to the Go version, which mutated the `*Notice`. However, I think you're right that it's better to make the implementation slightly trickier and keep the frozen=True as a good signal. I've switched this back to frozen, and updated `_notify` to use `dataclasses.replace()` instead.